### ISSUE TITLE
sunspot:solr:start can not create core container

### ISSUE THREAD

**Issue by urbanminingco** [sunspot/sunspot#980]
Hello, 

I'm trying to start solr using bundle exec sunspot:solr:run. However I get the following errors:

```
6650 ERROR (coreLoadExecutor-6-thread-1) [   x:test] o.a.s.c.CoreContainer Error creating core [test]: Error opening new searcher
org.apache.solr.common.SolrException: Error opening new searcher
	at org.apache.solr.core.SolrCore.<init>(SolrCore.java:820)
	at org.apache.solr.core.SolrCore.<init>(SolrCore.java:659)
	at org.apache.solr.core.CoreContainer.create(CoreContainer.java:723)
	at org.apache.solr.core.CoreContainer$1.call(CoreContainer.java:443)
	at org.apache.solr.core.CoreContainer$1.call(CoreContainer.java:434)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.apache.solr.common.util.ExecutorUtil$MDCAwareThreadPoolExecutor$1.run(ExecutorUtil.java:210)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: org.apache.solr.common.SolrException: Error opening new searcher
	at org.apache.solr.core.SolrCore.openNewSearcher(SolrCore.java:1662)
	at org.apache.solr.core.SolrCore.getSearcher(SolrCore.java:1773)
	at org.apache.solr.core.SolrCore.initSearcher(SolrCore.java:916)
	at org.apache.solr.core.SolrCore.<init>(SolrCore.java:793)
	... 9 more
Caused by: java.io.IOException: Unable to unmap the mapped buffer: MMapIndexInput(path="{my-project-path}/solr/test/data/index/segments_1")
	at org.apache.lucene.store.MMapDirectory$2.freeBuffer(MMapDirectory.java:329)
	at org.apache.lucene.store.ByteBufferIndexInput.freeBuffer(ByteBufferIndexInput.java:376)
	at org.apache.lucene.store.ByteBufferIndexInput.close(ByteBufferIndexInput.java:355)
	at org.apache.lucene.store.BufferedChecksumIndexInput.close(BufferedChecksumIndexInput.java:60)
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:459)
	at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:909)
	at org.apache.solr.update.SolrIndexWriter.<init>(SolrIndexWriter.java:78)
	at org.apache.solr.update.SolrIndexWriter.create(SolrIndexWriter.java:65)
	at org.apache.solr.update.DefaultSolrCoreState.createMainIndexWriter(DefaultSolrCoreState.java:274)
	at org.apache.solr.update.DefaultSolrCoreState.getIndexWriter(DefaultSolrCoreState.java:117)
	at org.apache.solr.core.SolrCore.openNewSearcher(SolrCore.java:1631)
	... 12 more
```
I have tried, deleting the write.lock on all the solr subdirectories, deleting the solr folder, and re running the installation, downgrading to java 12. But still I haven't been able to solve the issue. I'm wondering what else can try? 

The gem version is: 2.5.0
running java --version gives the following: 
```
java --version
java 12.0.1 2019-04-16
Java(TM) SE Runtime Environment (build 12.0.1+12)
Java HotSpot(TM) 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)
```

Thanks!

**Comment by mlh758** [comment]
The included version of Solr is _very_ old and I don't recommend you use it, even for development. There is a [docker container](https://hub.docker.com/_/solr/) for Solr that is just as easy to use.

**Comment by vakuum** [comment]
@urbanminingco: For local development you can use Java 8 to start the included Solr with the rake tasks `sunspot:solr:start` or `sunspot:solr:run`. As @mlh758 already noted: You should not use `sunspot_solr` for production: https://github.com/sunspot/sunspot/issues/954

The included version is based on Solr 5.3.1 from 2015:

* https://github.com/sunspot/sunspot/tree/v2.5.0/sunspot_solr/solr/server/
* https://github.com/sunspot/sunspot/tree/v2.5.0/sunspot_solr/solr/server/solr-webapp/webapp/WEB-INF/lib/solr-core-5.3.1.jar
* https://github.com/apache/lucene-solr/tree/releases/lucene-solr/5.3.1/
* https://github.com/apache/lucene-solr/tree/releases/lucene-solr/5.3.1/solr/server

The current LTS versions of Java are Java 8 and Java 11: https://en.wikipedia.org/wiki/Java_version_history

It works with Java 8:

```
$ export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/jre
$ export PATH=$JAVA_HOME/bin:$PATH

$ java -version
openjdk version "1.8.0_282"
OpenJDK Runtime Environment (build 1.8.0_282-8u282-b08-0ubuntu1~18.04-b08)
OpenJDK 64-Bit Server VM (build 25.282-b08, mixed mode)

$ bundle exec rake sunspot:solr:run
...
1135 INFO  (coreLoadExecutor-6-thread-2) [   x:default] o.a.s.c.CoreContainer registering core: default
...
1143 INFO  (coreLoadExecutor-6-thread-1) [   x:test] o.a.s.c.CoreContainer registering core: test
...
1143 INFO  (coreLoadExecutor-6-thread-3) [   x:development] o.a.s.c.CoreContainer registering core: development
...
^C
```

It doesn't work with Java 11:

```
$ export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64
$ export PATH=$JAVA_HOME/bin:$PATH

$ java -version
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+9-Ubuntu-0ubuntu1.18.04)
OpenJDK 64-Bit Server VM (build 11.0.10+9-Ubuntu-0ubuntu1.18.04, mixed mode, sharing)

$ bundle exec rake sunspot:solr:run
...
1170 ERROR (coreLoadExecutor-6-thread-1) [   x:test] o.a.s.c.CoreContainer Error creating core [test]: Error opening new searcher
org.apache.solr.common.SolrException: Error opening new searcher
        at org.apache.solr.core.SolrCore.<init>(SolrCore.java:820)
        at org.apache.solr.core.SolrCore.<init>(SolrCore.java:659)
        at org.apache.solr.core.CoreContainer.create(CoreContainer.java:723)
        at org.apache.solr.core.CoreContainer$1.call(CoreContainer.java:443)
        at org.apache.solr.core.CoreContainer$1.call(CoreContainer.java:434)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.apache.solr.common.util.ExecutorUtil$MDCAwareThreadPoolExecutor$1.run(ExecutorUtil.java:210)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.apache.solr.common.SolrException: Error opening new searcher
        at org.apache.solr.core.SolrCore.openNewSearcher(SolrCore.java:1662)
        at org.apache.solr.core.SolrCore.getSearcher(SolrCore.java:1773)
        at org.apache.solr.core.SolrCore.initSearcher(SolrCore.java:916)
        at org.apache.solr.core.SolrCore.<init>(SolrCore.java:793)
        ... 9 more
Caused by: java.io.IOException: Unable to unmap the mapped buffer: MMapIndexInput(path="/home/clemens/development/ruby/mhmr/standard/solr/test/data/index/segments_1")
        at org.apache.lucene.store.MMapDirectory$2.freeBuffer(MMapDirectory.java:329)
        at org.apache.lucene.store.ByteBufferIndexInput.freeBuffer(ByteBufferIndexInput.java:376)
        at org.apache.lucene.store.ByteBufferIndexInput.close(ByteBufferIndexInput.java:355)
        at org.apache.lucene.store.BufferedChecksumIndexInput.close(BufferedChecksumIndexInput.java:60)
        at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:459)
        at org.apache.lucene.index.IndexWriter.<init>(IndexWriter.java:909)
        at org.apache.solr.update.SolrIndexWriter.<init>(SolrIndexWriter.java:78)
        at org.apache.solr.update.SolrIndexWriter.create(SolrIndexWriter.java:65)
        at org.apache.solr.update.DefaultSolrCoreState.createMainIndexWriter(DefaultSolrCoreState.java:274)
        at org.apache.solr.update.DefaultSolrCoreState.getIndexWriter(DefaultSolrCoreState.java:117)
        at org.apache.solr.core.SolrCore.openNewSearcher(SolrCore.java:1631)
        ... 12 more
Caused by: java.lang.IllegalAccessException: class org.apache.lucene.store.MMapDirectory$2$1 cannot access class jdk.internal.ref.Cleaner (in module java.base) because module java.base does not export jdk.internal.ref to unnamed module @5b73ad56
        at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
        at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
        at java.base/java.lang.reflect.Method.invoke(Method.java:558)
        at org.apache.lucene.store.MMapDirectory$2$1.run(MMapDirectory.java:322)
        at org.apache.lucene.store.MMapDirectory$2$1.run(MMapDirectory.java:314)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at org.apache.lucene.store.MMapDirectory$2.freeBuffer(MMapDirectory.java:314)
        ... 22 more
...
^C
```

**Comment by mlh758** [comment]
I was suggesting you not even use it for development purposes. I doubt you would want to base your config around such an old Solr version or have to have substantially different versions and configurations between development and production.

Also, I'm going to close this now since the original person never replied.
